clients/privacy-center/components/consent/NoticeDrivenConsent.tsx

```diff
@@ -0,0 +1,194 @@
+import { Divider, Stack, useToast } from "@fidesui/react";
+import React, { useEffect, useMemo, useState } from "react";
+import {
+  ConsentContext,
+  getConsentContext,
+  getOrMakeFidesCookie,
+  saveFidesCookie,
+} from "fides-js";
+import { useAppSelector } from "~/app/hooks";
+import {
+  selectCurrentConsentPreferences,
+  selectExperienceRegion,
+  selectPrivacyExperience,
+  useUpdatePrivacyPreferencesUnverifiedMutation,
+} from "~/features/consent/consent.slice";
+import {
+  getGpcStatusFromNotice,
+  transformUserPreferenceToBoolean,
+} from "~/features/consent/helpers";
+
+import {
+  ConsentMethod,
+  ConsentOptionCreate,
+  PrivacyNoticeResponseWithUserPreferences,
+  PrivacyPreferencesRequest,
+  UserConsentPreference,
+} from "~/types/api";
+import { useRouter } from "next/router";
+import { inspectForBrowserIdentities } from "~/common/browser-identities";
+import { NoticeHistoryIdToPreference } from "~/features/consent/types";
+import { ErrorToastOptions, SuccessToastOptions } from "~/common/toast-options";
+import ConsentItem from "./ConsentItem";
+import SaveCancel from "./SaveCancel";
+
+const resolveConsentValue = (
```

Comment:
okay yes, it should go in `fides-js` because `makeConsentDefaults` will need it once it can support privacy notices! (which is part of this work: https://github.com/ethyca/fides/issues/3377)

Reply:
I'm going to leave it alone for now so that this PR doesn't mess with `fides-js`, but whichever one of us gets to https://github.com/ethyca/fides/issues/3377 should move this function over at that time :)